File: 27-remove-element/remove-element.py
class Solution(object):
    def removeElement(self, nums, val):
        """
        :type nums: List[int]
        :type val: int
        :rtype: int
        """
        s = 0
        e = len(nums)
        while (s < e):
            if (nums[s] == val):
                nums[s] = nums[e-1]
                e -= 1
            else:
                s += 1
        return s

File: 27-remove-element/remove-element.java
class Solution {
    public int removeElement(int[] nums, int val) {
        int le = nums.length;
        int k = nums.length;
        int ix = 0;
        while (ix < le) {
            if (nums[ix] != val) {
                ix++;
            } else {
                k--;
                nums[ix] = nums[le - 1];
                le--;
            }
        }

        return k;
    }
}